gms/impl/src/main/java/com/linkedin/metadata/resources/entity/AspectResource.java

```diff
@@ -46,9 +50,10 @@
       final VersionedAspect aspect = _entityService.getVersionedAspect(urn, aspectName, version);
       if (aspect == null) {
         throw RestliUtils.resourceNotFoundException();
+      } else {
+        validateRecord(aspect, HttpStatus.S_500_INTERNAL_SERVER_ERROR);
```

Comment:
I think Datahub should throw a 400 in this case- this is a bad request, not an internal server error

Reply:
Yes sir!